app/src/main/java/com/p2p/presentation/tuttifrutti/play/PlayTuttiFruttiFragment.kt

```diff
@@ -0,0 +1,120 @@
+package com.p2p.presentation.tuttifrutti.play
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import androidx.cardview.widget.CardView
+import androidx.core.view.children
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.android.material.textfield.TextInputLayout
+import com.p2p.R
+import com.p2p.databinding.FragmentPlayTuttiFruttiBinding
+import com.p2p.databinding.PlayCategoryItemBinding
+import com.p2p.presentation.basegame.BaseGameFragment
+import com.p2p.presentation.tuttifrutti.TuttiFruttiViewModel
+import com.p2p.presentation.tuttifrutti.create.categories.Category
+import com.p2p.utils.clear
+import com.p2p.utils.text
+
+class PlayTuttiFruttiFragment : BaseGameFragment<
+        FragmentPlayTuttiFruttiBinding,
+        TuttiFruttiPlayingEvents,
+        PlayTuttiFruttiViewModel,
+        TuttiFruttiViewModel>() {
+
+    override val gameViewModel: TuttiFruttiViewModel by activityViewModels()
+
+    override val viewModel: PlayTuttiFruttiViewModel by viewModels()
+
+    override val gameInflater: (LayoutInflater, ViewGroup?, Boolean) -> FragmentPlayTuttiFruttiBinding =
+        FragmentPlayTuttiFruttiBinding::inflate
+
+    override fun initValues() {
+        gameViewModel.startGame()
+    }
+
+    override fun initUI() {
+        super.initUI()
+        with(gameBinding) {
+            setUpCategoriesList(categoriesList)
+            finishRoundButton.setOnClickListener {
+                val values = textViews().map { it.text() }
+                val categoriesWithValues =
+                    gameViewModel.selectedCategories.value!!.zip(values).toMap()
+                viewModel.onEndRound(categoriesWithValues)
+            }
+        }
+    }
+
+    private fun textViews(): List<TextInputLayout> {
```

Comment:
```suggestion
    private fun getCategoriesTextInputs(): List<TextInputLayout> {
```
Para que se entienda qué está trayendo 😅 Igual si usás el mapa que te dije arriba esto volaría

Reply:
done, no volo poruqe se usa para los errores